src/libstore/filetransfer.cc

```diff
@@ -77,6 +78,13 @@ struct curlFileTransfer : public FileTransfer
 
         std::chrono::steady_clock::time_point startTime = std::chrono::steady_clock::now();
 
+#if NIX_WITH_S3_SUPPORT
+        // AWS SigV4 authentication data
+        bool isS3Request = false;
```

Comment:
```suggestion
        std::optional<ParsedS3URL> isS3Request;
```
this might be overkill for now, but it is a good lack of boolean blindness and may help later.